arch/arm/include/tls.h

```diff
@@ -97,11 +82,15 @@ static inline uint32_t up_getsp(void)
  *
  ****************************************************************************/
 
+#ifdef CONFIG_TLS_ALIGNED
 static inline FAR struct tls_info_s *up_tls_info(void)
 {
   DEBUGASSERT(!up_interrupt_context());
   return TLS_INFO((uintptr_t)up_getsp());
```

Comment:
Should we remove up_tls_info and tls.h but move up_getsp to arch.h? So tls_get_info can implement align/unalign in the common place.

Reply:
I don't think so.  arch/arm/include/tls.h and arch/sim/include/tls.h implement this in very different ways.  sim uses a builtin.

Reply:
I think it is good to keep the architecture-specific hook to handle any future differences we encounter.  There is no benefit to removing it.  There is benefit to keeping it.

Reply:
Yes, but sim up_getsp can call __builtin_frame_address for us, right?
The problem is that:
1.up_getsp is implemented by most arch
2.the aglined stack allocation also support by several arch
3.but tls.h is just implemented on arm/sim
or1k is one of case we need modify too, some arch(riscv/xtensa) implment the partial tls which is better to remove or enhance.